integration/hibernate-4/src/main/java/com/blazebit/persistence/impl/hibernate/Hibernate4Integrator.java

```diff
@@ -36,7 +37,8 @@ public void integrate(Configuration configuration, SessionFactoryImplementor ses
 				// TODO: check that no collections are mapped
 			}
 		}
-		CommonHibernateUtils.setCollectionPersisterClass(configuration.getClassMappings(), CustomOneToManyPersister.class);
+
+		serviceRegistry.locateServiceBinding(PersisterClassResolver.class).setService(new CustomPersisterClassResolver());
```

Comment:
The only thing that the resolver does not take into account as far as I can see is printing out a warning if the user supplies a custom persister. What do you think, is it better to print the warning or to write documentation about how the fetch support works and let the user figure out that the custom persister is the reason why it doesn't work?